sycl/test/on-device/sub_group_algorithms/any_of.cpp

```diff
@@ -0,0 +1,74 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -I . -o %t.out
```

Comment:
the file can be combined with all_of.cpp because the only difference is function used. So we can move common part to header and got less code duplication.

Reply:
between the predicates, the different calls to `any_of_group`, `joint_any_of`, (and the variations for `all_of`) and then checking via `std::any_of/all_of`  that starts to get very meta, which means the test will be difficult to debug or work with if an error shows up at some later date. I think it might be best to keep the files separate.